src/lido/LidoOracle.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {IERC20} from "../interfaces/IERC20.sol";
+import {IWSTETH} from "./IWSTETH.sol";
+import {BoringERC20} from "../libraries/BoringERC20.sol";
+
+contract LidoOracle {
+    using BoringERC20 for IERC20;
+    error OracleUnsupportedPair();
```

Comment:
To abide by the spec, this error should be `error OracleUnsupportedPair(address base, address quote);`